snuba/datasets/entity.py

```diff
@@ -0,0 +1,91 @@
+from typing import Mapping, Optional, Sequence
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.clickhouse.escaping import escape_identifier
+from snuba.datasets.plans.query_plan import ClickhouseQueryPlanBuilder
+from snuba.datasets.storage import Storage, WritableStorage, WritableTableStorage
+from snuba.query.extensions import QueryExtension
+from snuba.query.logical import Query
+from snuba.query.parsing import ParsingContext
+from snuba.query.processors import QueryProcessor
+from snuba.util import qualified_column
+
+
+class Entity(object):
```

Comment:
Making this abstract makes sense to me. It's not right now because Dataset is not an abstract class. I'll change entities now and Dataset should probably change too in a separate PR.